### ISSUE TITLE
Built in CCJoystick class.

### ISSUE THREAD

**Issue by slembcke** [cocos2d/cocos2d-objc#477]
CCButton works well enough for virtual gamepad buttons, but I think we should add a default type for virtual joysticks and as well. I've implemented my own several times now.

I also pretty interested in adding iOS 7 controller support. I'm a sucker for physical buttons. It would be pretty easy to just tell Cocos to associate a CCJoystick with a particular physical joystick.

**Comment by cocojoe** [comment]
Sounds good, is this a nice to have for 3.1?

On 6 January 2014 23:17, slembcke notifications@github.com wrote:

> CCButton works well enough for virtual gamepad buttons, but I think we
> should add a default type for virtual joysticks and as well. I've
> implemented my own several times now.
> 
> I also pretty interested in adding iOS 7 controller support. I'm a sucker
> for physical buttons. It would be pretty easy to just tell Cocos to
> associate a CCJoystick with a particular physical joystick.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/cocos2d/cocos2d-iphone/issues/477
> .

## 

Martin

**Comment by slembcke** [comment]
Yeah. I consider it very low priority, but also very easy. It's maybe a day of work to make a pretty fancy customizable one.

**Comment by billhollings** [comment]
To save yourself the day of work, just rip the `Joystick` class out of the **cocos3d** `CC3DemoMashUp` or `CC3Performance` demo apps.

It's a pure _cocos2d_ component (no 3D code), and it is "fancy customizable"

It's in `Joystick.h/m`. I can email you the files if you don't want to pull **cocos3d**.
